src/main/java/com/server/CLIFlagParser.java

```diff
@@ -0,0 +1,27 @@
+package com.server;
+
+import java.io.IOException;
+
+public class CLIFlagParser {
+
+    public ServerConfig parse(String[] args) throws IOException {
+        int portNumber;
+        String directory;
+
+        try{
+            if(args[0].equals("-p") && args[2].equals("-d")){
```

Comment:
How do we feel about having "magic" strings `-p` and `-d`? Is it worth considering extracting these?